autocodesign/projectmanager/projecthelper.go

```diff
@@ -202,44 +225,171 @@ func (p *ProjectHelper) ProjectTeamID(config string) (string, error) {
+		return settingsList[0].settings, nil
+	}
+
+	p.Logger.Debugf("buildSettings: Workspace target build settings: %+v", settingsList[0])
+	p.Logger.Debugf("buildSettings: Project target build settings: %+v", settingsList[1])
+	p.Logger.Debugf("buildSettings: Multiple build settings found for target (%s), returning the project one", targetName)
+	return settingsList[1].settings, nil
+}
+
+func (p *ProjectHelper) buildSettingForKey(targetName, conf string, key string, customOptions ...string) (string, error) {
+	settingsList, err := p.cachedBuildSettings(targetName, conf, customOptions...)
+	if err != nil {
+		var basePath string
+		if len(settingsList) > 0 {
+			basePath = settingsList[0].basePath
+		}
+		return "", fmt.Errorf("failed to fetch target (%s) build settings for project (%s): %s", targetName, basePath, err)
+	}
+
+	wsSettings := settingsList[0].settings
+	wsValue, err := wsSettings.String(key)
+	if err != nil {
+		return wsValue, err
+	}
+
+	if len(settingsList) == 1 {
+		return wsValue, nil
+	}
+
+	projectSettings := settingsList[1].settings
+	projectValue, err := projectSettings.String(key)
+	if err != nil {
+		p.Logger.Errorf("buildSettings: Failed to fetch project build setting for key (%s): %s", key, err)
+		p.Logger.Printf("buildSettings: Returning workspace value for key (%s): %s", key, wsValue)
+		return wsValue, nil
+	}
+
+	if projectValue != wsValue {
+		p.Logger.Errorf("buildSettings: Conflicting values for build setting %s: '%s' (workspace) vs '%s' (project)", key, wsValue, projectValue)
+		// Return alternate value to be consistent with old project based target build setting fetch
+		p.Logger.Printf("buildSettings: Returning project value for key (%s): %s", key, projectValue)
+		return projectValue, nil
 	}
-	p.buildSettingsCache[name] = targetCache
+	p.Logger.Debugf("buildSettings: Matching values for workspace and project build setting %s: '%s'", key, wsValue)
 
-	return settings, nil
+	return wsValue, err
+}
+
+func (p *ProjectHelper) buildSettingPathForKey(targetName, conf string, key string, customOptions ...string) (string, error) {
+	settingsList, err := p.cachedBuildSettings(targetName, conf, customOptions...)
+	if err != nil {
+		var basePath string
+		if len(settingsList) > 0 {
+			basePath = settingsList[0].basePath
+		}
+		return "", fmt.Errorf("failed to fetch target (%s) build settings for project (%s): %s", targetName, basePath, err)
+	}
+
+	wsSettings := settingsList[0]
+	wsValue, err := wsSettings.settings.String(key)
+	if err != nil {
+		return wsValue, err
+	}
+
+	if pathutil.IsRelativePath(wsValue) {
+		wsValue = filepath.Join(path.Dir(wsSettings.basePath), wsValue)
+	}
+
+	if len(settingsList) == 1 {
+		return wsValue, nil
+	}
+
+	projectSettings := settingsList[1]
+	projectValue, err := projectSettings.settings.String(key)
+	if err != nil {
+		return projectValue, err
+	}
+
+	if pathutil.IsRelativePath(projectValue) {
+		projectValue = filepath.Join(path.Dir(projectSettings.basePath), projectValue)
```

Comment:
nit: use `filepath.Dir()` for consistency

Reply:
Fixed.